tests/OTAnalytics/plugin_ui/test_cli.py

```diff
@@ -391,57 +433,283 @@ def cli_dependencies(self) -> dict[str, Any]:
+            FillZerosExporterFactory(
+                AddSectionInformationExporterFactory(SimpleExporterFactory())
+            ),
+        )
+        tracks_metadata = TracksMetadata(track_repository)
+        videos_metadata = VideosMetadata()
+        export_tracks = CsvTrackExport(
+            track_repository, tracks_metadata, videos_metadata
+        )
+        export_road_user_assignments = ExportRoadUserAssignments(
+            event_repository=event_repository,
+            flow_repository=flow_repository,
+            create_events=create_events,
+            assigner=assigner,
+            exporter_factory=SimpleRoadUserAssignmentExporterFactory(
+                section_repository, get_all_tracks
+            ),
+        )
+        return {
+            self.EVENT_REPOSITORY: event_repository,
+            self.ADD_SECTION: AddSection(section_repository),
+            self.GET_ALL_SECTIONS: GetAllSections(section_repository),
+            self.ADD_FLOW: AddFlow(flow_repository),
+            self.CREATE_EVENTS: create_events,
+            self.EXPORT_COUNTS: export_counts,
+            self.EXPORT_TRACKS: export_tracks,
+            self.PROVIDE_EVENTLIST_EXPORTER: provide_available_eventlist_exporter,
+            self.APPLY_CLI_CUTS: apply_cli_cuts,
+            self.ADD_ALL_TRACKS: add_all_tracks,
+            self.GET_ALL_TRACK_IDS: get_all_track_ids,
+            self.CLEAR_ALL_TRACKS: clear_all_tracks,
+            self.TRACKS_METADATA: tracks_metadata,
+            self.VIDEOS_METADATA: videos_metadata,
+            self.EXPORT_ROAD_USER_ASSIGNMENT: export_road_user_assignments,
+        }
+
+    @pytest.fixture
+    def cli_stream_dependencies(
+        self, cli_dependencies: dict[str, Any]
+    ) -> dict[str, Any]:
+        dependencies = cli_dependencies
+        tracks_metadata = dependencies[self.TRACKS_METADATA]
+        videos_metadata = dependencies[self.VIDEOS_METADATA]
+        return {
+            self.TRACK_PARSER: StreamOttrkParser(
+                PythonStreamDetectionParser(
+                    ByMaxConfidence(), DEFAULT_TRACK_LENGTH_LIMIT
+                ),
+                registered_tracks_metadata=[tracks_metadata],
+                registered_videos_metadata=[videos_metadata],
+                chunk_size=2,
+            ),
+            **dependencies,
+        }
+
+    @pytest.fixture
+    def cli_bulk_dependencies(self, cli_dependencies: dict[str, Any]) -> dict[str, Any]:
+        dependencies = cli_dependencies
+        tracks_metadata: TracksMetadata = dependencies[self.TRACKS_METADATA]
+
+        return {
+            self.TRACK_PARSER: OttrkParser(
+                PythonDetectionParser(
+                    ByMaxConfidence(),
+                    tracks_metadata._track_repository,
+                    PygeosTrackGeometryDataset.from_track_dataset,
+                    DEFAULT_TRACK_LENGTH_LIMIT,
+                ),
+            ),
+            self.PROGRESSBAR: NoProgressbarBuilder(),
```

Comment:
Why is `self.PROGRESSBAR` only set for bulk_dependencies?

Reply:
For OTAnalyticsStreamCli, the progressbar shows progress in terms of processed files, as the number of tracks is not known from the lazy stream.
Files are being processed in the StreamOttrkParser implementation, hence the progressbar is a constructor argument of this object and not cli (since the parser itself is a constructor argument of cli)